sssd_test_framework/utils/tools.py

```diff
@@ -634,19 +638,27 @@ def initgroups(self, name: str) -> InitgroupsEntry:
         """
         return self.__exec(InitgroupsEntry, "initgroups", name)
 
-    def netgroup(self, name: str) -> NetgroupEntry | None:
+    def netgroup(self, name: str, service: str | None = None) -> NetgroupEntry | None:
         """
         Call ``getent netgroup $name``
 
         :param name: Netgroup name.
         :type name: str
+        :param service: Service used. | Defaults to None
+        :type service: str
         :return: Netgroup data, None if not found
         :rtype: NetgroupEntry | None
         """
-        return self.__exec(NetgroupEntry, "netgroup", name)
+        return self.__exec(NetgroupEntry, "netgroup", name, service)
+
+    def __exec(self, cls, cmd: str, name: str | int, service: str | None = None) -> Any:
+        args = []
+
+        if service is not None:
+            args = ["-s", service]
+
+        command = self.host.ssh.exec(["getent", *args, cmd, name], raise_on_error=False)
 
```

Comment:
Please, remove this blank line. It is good to keep assignement and its check together.